lib/modules/datasource/hexpm-bob/index.ts

```diff
@@ -0,0 +1,180 @@
+import { logger } from '../../../logger';
+import { ExternalHostError } from '../../../types/errors/external-host-error';
+import { cache } from '../../../util/cache/package/decorator';
+import { HttpError } from '../../../util/http';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import { PackageType, datasource, defaultRegistryUrl } from './common';
+
+export class HexpmBobDatasource extends Datasource {
+  static readonly id = datasource;
+
+  constructor() {
+    super(datasource);
+  }
+
+  override readonly customRegistrySupport = true;
+
+  override readonly defaultRegistryUrls = [defaultRegistryUrl];
+
+  override readonly caching = true;
+
+  @cache({
+    namespace: `datasource-${datasource}`,
+    key: ({ registryUrl, packageName }: GetReleasesConfig) =>
+      `${registryUrl ?? defaultRegistryUrl}:${packageName}`,
+  })
+  async getReleases({
+    registryUrl,
+    packageName,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    const packageType = this.getPackageType(packageName);
+
+    if (!packageType) {
+      return null;
+    }
+
+    logger.trace(
+      { registryUrl: registryUrl ?? defaultRegistryUrl, packageName },
+      `fetching hex.pm bob ${packageName} release`
+    );
+
+    const url = `${
+      registryUrl ?? defaultRegistryUrl
+    }/builds/${packageName}/builds.txt`;
+
+    const result: ReleaseResult = {
+      releases: [],
+      ...this.getPackageDetails(packageType),
+    };
+    try {
+      const { body } = await this.http.get(url);
+      result.releases = body
+        .split('\n')
+        .map((line) => line.trim())
+        .filter((line) => line !== '')
+        .map((line) => {
+          const [version, gitRef, buildDate, newDigest] = line.split(' ');
+
+          return {
+            changelogUrl: this.changelogUrl(version, packageType),
+            downloadUrl: `${
+              registryUrl ?? defaultRegistryUrl
+            }builds/${packageName}/${version}.tar.gz`,
```

Comment:
is that used? i don't think we need that, so should removed.

Reply:
@viceice Currently not, but theoretically the data source could be used in a different context than the `asdf` manager.

Then the `downloadUrl` could be used, no?

Reply:
no, we don't use any datasource to download binary assets.